src/sonic-daemon-base/sonic_daemon_base/daemon_base.py

```diff
@@ -151,13 +152,17 @@ def get_path_to_port_config_file(self):
         # Get platform and hwsku path
         (platform_path, hwsku_path) = self.get_path_to_platform_and_hwsku()
 
-        # First check for the presence of the new 'port_config.ini' file
-        port_config_file_path = "/".join([hwsku_path, PORT_CONFIG])
+        # First check for the presence of the new 'platform.json' file
+        port_config_file_path = "/".join([hwsku_path, PLATFORM_CAP])
         if not os.path.isfile(port_config_file_path):
-            # port_config.ini doesn't exist. Try loading the legacy 'portmap.ini' file
-            port_config_file_path = "/".join([hwsku_path, PORTMAP])
+
+            # platform.json doesn't exist. Try loading the legacy 'port_config.ini' file
+            port_config_file_path = "/".join([hwsku_path, PORT_CONFIG])
             if not os.path.isfile(port_config_file_path):
-                raise IOError("Failed to detect port config file: %s" % (port_config_file_path))
+                # port_config.ini doesn't exist. Try loading the legacy 'portmap.ini' file
```

Comment:
The legacy 'portmap.ini' file has been deprecated for over 3 years at this point. I think it is safe to remove all code pertaining to this file.

@lguohan: do you agree? If so, should that change be in a separate PR, or do you think it's OK to be included in this one?

Reply:
i agree.

Reply:
Should I remove it in this PR?

Reply:
Yes, please.

Reply:
#done